### PR TITLE
fix issue #60 (slave fails going to OP: fatal sync error)

### PR DESCRIPTION
**Pull request by Apollo3zehn**
- reset activation register in ecx_set_slaves_to_default

### CODE REVIEW

File: soem/ethercattype.h [hunk 1]
```diff
@@ -86,9 +86,9 @@ extern "C"
 /** datagram type EtherCAT */
 #define EC_ECATTYPE        0x1000
 /** number of frame buffers per channel (tx, rx1 rx2) */
-#define EC_MAXBUF          16
+#define EC_MAXBUF          256
```
Comment (nakarlsson) [inline soem/ethercattype.h]: Leave ethercattype.h as is, changes here affects all plattforms. I've written an #65 to make it possible to override values.  256 buffers don't make sense on small embedded systems.

File: osal/win32/osal.c [hunk 2]
```diff
@@ -48,6 +48,24 @@ static double qpc2usec;
 
 #define USECS_PER_SEC     1000000
 
+int osal_getabsolutetimeofday(struct timeval *tv, struct timezone *tz)
```
Comment (nakarlsson) [inline osal/win32/osal.c]: How does this function get affected by system time?  For the stack the most important task is to work as a stop watch and monitor when we have waited or tried long enough. It is kind of legacy from Linux that it is named "gettimeofday".

**Comment by Apollo3zehn** [comment]
Sorry, I thought that only my initial commit will be included in the pull request. The following commits were indeed only intended for my fork.

Is it better to recreate this pull request or are you able to deny all changes except those in ethercatconfig.c ?

File: osal/win32/osal.c (hunk 2 above)
Comment (Apollo3zehn) [inline osal/win32/osal.c]: This function provides a new system time value every 1-15 ms (depending on the settings made with [timeBeginPeriod](https://msdn.microsoft.com/en-us/library/windows/desktop/dd757624(v=vs.85).aspx)). On Windows 8 and above, there is the new function [GetSystemTimePreciseAsFileTime](https://msdn.microsoft.com/en-us/library/windows/desktop/hh706895(v=vs.85).aspx) that provides the system time with much higher precision (< 1 µs). 

My version of osal_getabsolutetimeofday is only a workaround to initialize DC with an absolute time but needs some modifications before publishing. I would prefer to create a separate pull request for this issue. As I wrote below, I mistakenly commited serveral changes to this pull request.

**Comment by nakarlsson** [comment]
Please recreate the pull request.

**Comment by Apollo3zehn** [comment]
created new pull request with correct changes

**Pull request closed without merging**